### PR TITLE
feat: log request and response bodies independently from ktor

### PR DESCRIPTION
**Pull request by osama-salman99**
* [x] Code is up-to-date with the `main` branch.
* [x] You've successfully built and run the tests locally.
* [ ] There are new or updated unit tests validating the change.

### :pencil: Description
- Log request and response bodies independently from ktor (fixes duplicate response body logs)

### :link: Related Issues
N/A

### CODE REVIEW

File: core/src/main/kotlin/com/expediagroup/sdk/core/plugin/logging/RequestBodyLogger.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    companion object Plugin : HttpClientPlugin<RequestBodyLoggerConfig, RequestBodyLogger> {
+        override val key: AttributeKey<RequestBodyLogger> = AttributeKey(LoggerName.REQUEST_BODY_LOGGER)
+
+        override fun install(
+            plugin: RequestBodyLogger,
+            scope: HttpClient
+        ) {
+            scope.sendPipeline.intercept(HttpSendPipeline.Monitoring) {
+                val body: String = getBody()
+                plugin.log.info(LoggingMessageProvider.getRequestBodyMessage(body, context.headers.getTransactionId()))
```
Comment (anssari1) [inline core/src/main/kotlin/com/expediagroup/sdk/core/plugin/logging/RequestBodyLogger.kt]: change to debug

File: core/src/main/kotlin/com/expediagroup/sdk/core/plugin/logging/ResponseBodyLogger.kt [hunk 2]
```diff
@@ -46,11 +48,13 @@ class ResponseBodyLogger {
                 val response: HttpResponse = context.response
                 val byteReadChannel: ByteReadChannel = if (response.contentEncoding().equals(HeaderValue.GZIP)) scope.decode(response.content) else response.content
                 val body: String = byteReadChannel.readRemaining().readText()
-                plugin.log.info(LoggingMessageProvider.getResponseBodyMessage(body))
+                plugin.log.info(LoggingMessageProvider.getResponseBodyMessage(body, response.headers.getTransactionId()))
```
Comment (anssari1) [inline core/src/main/kotlin/com/expediagroup/sdk/core/plugin/logging/ResponseBodyLogger.kt]: change to debug

**Review by anssari1 — DISMISSED** [review]

File: core/src/main/kotlin/com/expediagroup/sdk/core/plugin/logging/RequestBodyLogger.kt (hunk 1 above)
Comment (osama-salman99) [reply core/src/main/kotlin/com/expediagroup/sdk/core/plugin/logging/RequestBodyLogger.kt]: Done ✅

File: core/src/main/kotlin/com/expediagroup/sdk/core/plugin/logging/ResponseBodyLogger.kt (hunk 2 above)
Comment (osama-salman99) [reply core/src/main/kotlin/com/expediagroup/sdk/core/plugin/logging/ResponseBodyLogger.kt]: Done ✅

**Review by anssari1 — APPROVED** [review]

**Pull request merged**